### PR TITLE
Fix building Python bindings on Mac

### PR DESCRIPTION
**Pull request by ManifoldFR**
## Description

This PR fixes building the Python on Mac.

- ~switch to eigenpy and include the `<eigenpy/eigenpy.hpp>` header~
- use the `PYTHON_EXT_SUFFIX` as a SUFFIX target property on the bindings to ensure Python on macOS properly detects the compiled module
- use the namespace-alias `namespace bp = boost::python`

## How I Tested

By building on a Mac Studio Ultra M1 and importing the module.

## I fulfilled the following requirements

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

**Comment by ManifoldFR** [comment]
A similar PR will be forthcoming on odri_control_interface.

File: sdk/master_board_sdk/CMakeLists.txt [hunk 1]
```diff
@@ -110,12 +102,14 @@ if(BUILD_PYTHON_INTERFACE)
...
   add_library(${PYWRAP} SHARED ${${PYWRAP}_SOURCES} ${${PYWRAP}_HEADERS})
-  target_include_directories(${PYWRAP} SYSTEM PRIVATE ${PYTHON_INCLUDE_DIR})
-  target_link_libraries(${PYWRAP} ${PROJECT_NAME})
-  target_link_boost_python(${PYWRAP})
-  get_relative_rpath(${${PYWRAP}_INSTALL_DIR} ${PYWRAP}_INSTALL_RPATH)
-  set_target_properties(${PYWRAP} PROPERTIES INSTALL_RPATH
-                                             "${${PYWRAP}_INSTALL_RPATH}")
+  set_target_properties(${PYWRAP} PROPERTIES SUFFIX ${PYTHON_EXT_SUFFIX})
+  target_link_libraries(${PYWRAP} PUBLIC ${PROJECT_NAME} eigenpy::eigenpy)
+  if(UNIX AND NOT APPLE)
```
Comment (nim65s) [inline sdk/master_board_sdk/CMakeLists.txt:107]: This should also work on apple, ref. https://github.com/stack-of-tasks/eigenpy/blob/master/python/CMakeLists.txt#L51

**Review by nim65s — COMMENTED** [review]
This is fine for me, but adding a dependency to eigenpy is a breaking change, and is a non-neglible extra burden for folks discovering ODRI who are not already pinocchio users.

I would like to know @thomasfla views on this before this is merged.

**Comment by nim65s** [comment]
And actually I don't see the point of adding eigenpy dependency. Does this add any feature, or fix any issue ?

**Comment by ManifoldFR** [comment]
My thinking was that it had useful fixes for compiling on Mac and linking to Boost.Python.

I'll look into compiling on Mac without resorting to depending on eigenpy.

**Comment by nim65s** [comment]
I have an example project with boost-python and macos CI here: https://github.com/Gepetto/example-adder. It is currently not in good shape, but that was working at some point, and I'll try to update it asap.

**Comment by ManifoldFR** [comment]
> I have an example project with boost-python and macos CI here: https://github.com/Gepetto/example-adder. It is currently not in good shape, but that was working at some point, and I'll try to update it asap.

Thanks a lot, I'll take a gander!

File: sdk/master_board_sdk/CMakeLists.txt (hunk 1 above)
Comment (ManifoldFR) [reply sdk/master_board_sdk/CMakeLists.txt:107]: Oh alright ! I was looking at aligator/proxsuite-nlp as reference. I'll update them and here.

**Comment by ManifoldFR** [comment]
Would you think it'd be worth it to also add CI for macOS for this repo ?

**Comment by ManifoldFR** [comment]
> I have an example project with boost-python and macos CI here: https://github.com/Gepetto/example-adder. It is currently not in good shape, but that was working at some point, and I'll try to update it asap.

On my end I am unable to build `example-adder` on Mac. I am getting the same problem as building the master-board bindings (without resorting to eigenpy), a linker error about missing Python symbols (`_PyLong_FromLong` for instance).

I'm attaching the output of `make` for master-board.
[error.log](https://github.com/open-dynamic-robot-initiative/master-board/files/15349845/error.log)

**Comment by nim65s** [comment]
Yes, CI would be nice. To build on macos arm, I think you need to pass `-arch arm64`

**Comment by ManifoldFR** [comment]
> Yes, CI would be nice. To build on macos arm, I think you need to pass `-arch arm64`

Inside of the CXXFLAGS ?

**Comment by nim65s** [comment]
Yes and LDFLAGS. Or from CMake: `-DCMAKE_APPLE_SILICON_PROCESSOR=amd64`

**Comment by ManifoldFR** [comment]
> Yes and LDFLAGS. Or from CMake: `-DCMAKE_APPLE_SILICON_PROCESSOR=amd64`

Didn't change anything for me unfortunately.

**Comment by ManifoldFR** [comment]
Ok, so the trick was to _remove_ the `search_for_boost_python()` line, so in `TARGET_LINK_BOOST_PYTHON` we go through the [else branch](https://github.com/jrl-umi3218/jrl-cmakemodules/blob/5275edfd1a8c6d45e0f90177edad93ef15b2008f/boost.cmake#L295) that doesn't link to the `Boost::python{major.minor}` target.

**Pull request closed without merging**